usr/src/uts/common/fs/zfs/zfs_log.c

```diff
@@ -481,30 +478,26 @@ zfs_log_write(zilog_t *zilog, dmu_tx_t *tx, int txtype,
 	while (resid) {
 		itx_t *itx;
 		lr_write_t *lr;
-		ssize_t len;
+		itx_wr_state_t wr_state = write_state;
```

Comment:
The idea was to create local copy for the present chunk to not modify global state if we change it for this for this chunk.  Though changing the global one may also be possible, though I am not sure it would simplify the logic rather then tangling it more.